From: Kel Modderman <kel@otaku42.de>
Date: Fri, 5 Apr 2019 20:32:41 +0000
Subject: Execute the Debian test suite on `make check'.

===================================================================
---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index d4c991b..15ea28e 100644
--- a/Makefile
+++ b/Makefile
@@ -148,7 +148,8 @@ ifeq ($(ISSUSE),-DSUSE)
 	issuse=true tests/common
 #	issuse=true tests/suse
 else
-	tests/common
+	chmod a+rx debian/run-testsuite
+	debian/run-testsuite
 endif
 
 install:	$(TODO) 
