From: Mohit Bhakkad <mohit.bhakkad@gmail.com>
Date: Fri, 5 Apr 2019 20:32:41 +0000
Subject: fix FTBFS with gcc-5 and clang.

Bug-Debian: https://bugs.debian.org/777914
---
 listing.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/listing.h b/listing.h
index 4247968..7ff2f08 100644
--- a/listing.h
+++ b/listing.h
@@ -47,9 +47,9 @@ typedef unsigned int uint;
 # ifndef  asm
 #  define asm			__asm__
 # endif
-# ifndef  extension
-#  define extension		__extension__
-# endif
+#endif
+#ifndef  extension
+# define extension             __extension__
 #endif
 #ifndef  attribute
 # define attribute(attr)	__attribute__(attr)
